### ISSUE TITLE
[HOLD - Pusher discussion] [$1000] placeholder shows instead of profile pic on change default contact method

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19327]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app
2. from userA add userB in Contact methods
3. login as userB
4. change default contact method from userA
5. check the profile pic in userB

## Expected Result: 
should show profile pic
## Actual Result: 
shows placeholder instead of profile pic

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.16.5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/ce101c84-1dd0-443b-b758-e1e0af23f18e

https://github.com/Expensify/App/assets/43996225/53185a3f-48b3-4f56-bf20-15149a2725e1

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684317183625259

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011f0dcfe82a9e3b71</li>
        <li>Upwork Job ID: 1661875558946312192</li>
        <li>Last Price Increase: 2023-05-25</li>
    </ul>
</details>

**Comment by therealsujitk** [comment]
Duplicate - #18780

**Comment by sakluger** [comment]
Thanks @therealsujitk, this is a duplicate of that other issue, closing in favor of the other one.

**Comment by gadhiyamanan** [comment]
this is a different issue. in this issue, the profile pic is not shown for userB and in other issue is about changing the profile pic
cc: @sakluger 
![Screenshot 2023-05-22 at 9 54 32 AM](https://github.com/Expensify/App/assets/54790231/fe2779ca-d891-4140-bd3a-39b88eec085d)

**Comment by therealsujitk** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The avatar (and other details such as the display name) isn't being updated if the default contact method is updated from another device (without refreshing).

### What is the root cause of that problem?

The reason for this is because Pusher only updates the `ONYXKEYS.LOGIN_LIST` and `ONYXKEYS.SESSION`, it doesn't update the `ONYXKEYS.PERSONAL_DETAILS`.

This is how it is done for the same device setting the default contact method in the [`setContactMethodAsDefault()`](https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/libs/actions/User.js#L771) function.

https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/libs/actions/User.js#L795-L806

### What changes do you think we should make in order to solve the problem?


One way to solve this is to update `ONYXKEYS.PERSONAL_DETAILS` from pusher, however this can also be done on the front-end instead.

This is the function `subscribeToUserDeprecatedEvents()` that currently takes care of user updates to Onyx.

https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/libs/actions/User.js#L605-L619

We'll have to check for when `ONYXKEYS.SESSION` data is updated here, and update `ONYXKEYS.PERSONAL_DETAILS` accordingly.

> **Note:** Obviously, this will also have to be done in `subscribeToUserEventsUsingMultipleEventType()` for when the server switches to sending account updates using `Pusher.TYPE.MULTIPLE_EVENT_TYPE`.

### What alternative solutions did you explore? (Optional)
None.

**Comment by gadhiyamanan** [comment]
bump @sakluger

**Comment by sakluger** [comment]
Sorry about that - I also didn't read through the entire other issue, I just thought it was the same after reading the issue title and scanning the description. After taking a closer look, I agree that this is a bug, and I was also able to reproduce.

**Comment by greg-schroeder** [comment]
Un-assigning as Sasha has this for BZ :)

**Comment by Julesssss** [comment]
Awaiting proposals

**Comment by therealsujitk** [comment]
@Julesssss would just like to note that I've added a proposal [here](https://github.com/Expensify/App/issues/19327#issuecomment-1556788451).

**Comment by abdulrahuman5196** [comment]
@Julesssss  
@therealsujitk's proposal root cause is partially correct. 

But the real issue here is, 
1) In OpenApp API  response, for `personalDetails` list we only get the value for the current default email while secondary contact information is not present.
2) when we change the default contact method, we send pusher to update the session. 
3) But since personalDetails doesn't have the information for the new email set as default, it will not populate currentUserPersonalDetails here
https://github.com/Expensify/App/blob/main/src/components/withCurrentUserPersonalDetails.js#L38-L42
4) This leads to UI not showing data related to personal details. Its not just profile pic, if we open the settings page name and other information is also not populated.
5) This gets fixed only if we reload, then Open API returns the current default contact's personalDetails in the response which fixes this issue.


The proper fix IMO here is to return the secondary emails personalDetails as part of the OpenAPI response.
So this should be taken up internal or wait on different idea to solve this externally.

**Comment by therealsujitk** [comment]
@abdulrahuman5196 yeah that's *exactly* what I wrote. Since this is done on the client end itself for the device that is switching the default contact method, I suggested this can be solved on the client end with the same logic. Here we use the old personal details and update the `login` and `displayMethod`.

https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/libs/actions/User.js#L795-L806
